### PR TITLE
typechecker: Disallow storing void types in storage types

### PR DESCRIPTION
**Pull request by Lubrsi**
Since you cannot form a reference to void, attempting to use a `void` type in any of the storage types is invalid and will not compile.

### CODE REVIEW

File: tests/typechecker/array_with_void_values.jakt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+enum Bar {
+    Value
+}
+
+function main() {
+    let foo = Bar::Value();
```
Comment (linusg) [inline tests/typechecker/array_with_void_values.jakt:6]: No semicolon please

File: tests/typechecker/set_with_void_values.jakt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+enum Bar {
+    Value
+}
+
+function main() {
+    let foo = Bar::Value();
```
Comment (linusg) [inline tests/typechecker/set_with_void_values.jakt:6]: Likewise

File: tests/typechecker/dict_with_void_keys.jakt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+enum Bar {
+    Value
+}
+
+function main() {
+    let foo = Bar::Value();
```
Comment (linusg) [inline tests/typechecker/dict_with_void_keys.jakt:6]: Likewise

File: tests/typechecker/dict_with_void_values.jakt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+enum Bar {
+    Value
+}
+
+function main() {
+    let foo = Bar::Value();
```
Comment (linusg) [inline tests/typechecker/dict_with_void_values.jakt:6]: Likewise

File: tests/typechecker/tuple_with_void_values.jakt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+enum Bar {
+    Value
+}
+
+function main() {
+    let foo = Bar::Value();
```
Comment (linusg) [inline tests/typechecker/tuple_with_void_values.jakt:6]: Likewise

**Review by linusg — CHANGES_REQUESTED** [review]

**Comment by Lubrsi** [comment]
Done :+1:

**Review by linusg — APPROVED** [review]

**Pull request merged**